deps/src/singular.cpp

```diff
@@ -206,6 +206,72 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     }
     return old_value;
   });
+  // all of the global setters return the previous value
+  Singular.method("set_option", [](std::string opt, bool value, ring r) {
```

Comment:
Can't we just do
```suggestion
  Singular.method("set_option", [](std::string opt, bool value, ring r == NULL) {
```
and then get rid of the old `Singular.method("set_option", [](std::string opt, bool value) {` method  (or dually, and how it'll appear in the diff: modify the existing method to take an optional `ring r = NULL`).

Reply:
OK, well, not sure if that *exactly* would work, in the context of `Singular.method` (can closures have default values for arguments???). But then we could also just leave the third argument as non-optional, and solve this on the Julia level. Either by adjusting all `libSingular.set_option` callers to pass in `C_NULL` as third argument. Or, more elegantly, by adding a 2-arg Julia method for `libSingular.set_option`, so like `libSingular.set_option(a,b) = libSingular.set_option(a,b,C_NULL)`